akka-docs/src/test/java/jdocs/stream/operators/sourceorflow/MapAsyncs.java

```diff
@@ -99,20 +103,153 @@ private void runUnordered() {
     // #mapasyncunordered
   }
 
+  private void runPartitioned() {
+    // #mapAsyncPartitioned
+    Source<EntityEvent, NotUsed> eventsForEntities = // ...
+        // #mapAsyncPartitioned
+        Source.fromIterator(() -> Stream.iterate(1, i -> i + 1).iterator())
+            .throttle(1, Duration.ofMillis(1))
+            .statefulMapConcat(
+                () -> {
+                  final HashMap<Integer, Integer> countsPerEntity = new HashMap<Integer, Integer>();
+
+                  return (n) -> {
+                    int entityId;
+
+                    if (random.nextBoolean() || countsPerEntity.isEmpty()) {
+                      entityId = random.nextInt(n);
+                    } else {
+                      int m = random.nextInt(countsPerEntity.size()) + 1;
+                      Iterator<Integer> keysIterator = countsPerEntity.keySet().iterator();
+
+                      Integer selected = Integer.valueOf(-1);
+                      int i = 0;
+                      while (keysIterator.hasNext() && i < m) {
+                        if (i == m - 1) {
+                          selected = keysIterator.next();
+                        } else {
+                          keysIterator.next();
+                        }
+                        i++;
+                      }
+                      entityId = selected.intValue();
+                    }
+
+                    if (entityId < 0) {
+                      throw new AssertionError("entityId should be non-negative, was " + entityId);
+                    }
+
+                    int seqNr =
+                        countsPerEntity
+                                .getOrDefault(Integer.valueOf(entityId), Integer.valueOf(0))
+                                .intValue()
+                            + 1;
+                    countsPerEntity.put(Integer.valueOf(entityId), Integer.valueOf(seqNr));
+
+                    return Collections.singletonList(new EntityEvent(entityId, seqNr));
+                  };
+                })
+            .take(1000);
```

Comment:
Ok, this elaborate pretend-events-from-broker source is not actually part of the sample.

Reply:
Yeah, followed the pattern elsewhere in the file.